src/forms/FormBuilder.js

```diff
@@ -34,25 +34,43 @@ export default class FormBuilder extends Component {
 
   constructor(props) {
     super(props);
-    // An empty form with no changes is valid as 'saved'
-    this.saved = true;
-    this.state = { openDialog: false }
-  }
 
-  markAsUnsaved() {
-    this.saved = false;
+    this.state = {
+      openDialog: false,
+      saved: true
+    };
+
+    this.onPublishOptions = this.onPublishOptions.bind(this);
+    this.onSaveClick = this.onSaveClick.bind(this);
+    this.addToBottom = this.addToBottom.bind(this);
+    this.onFormTitleChange = this.onFormTitleChange.bind(this);
+    this.markAsUnsaved = this.markAsUnsaved.bind(this);
+    this.renderPreview = this.renderPreview.bind(this);
+    this.hookRoute = this.hookRoute.bind(this);
   }
 
-  hookRouter() {
-    this.context.router.setRouteLeaveHook(this.props.route, () => {
-      if (this.saved === false) {
-        return 'This form has unsaved changes. Are you sure you want to leave this page?';
-      }
+  markAsUnsaved() {
+    this.setState({
+      saved: false
     });
   }
 
   componentDidMount() {
-    this.hookRouter();
+    console.log('component did mount')
+    const { router } = this.context;
+    const { route } =  this.props;
+
+    router.setRouteLeaveHook(route, this.hookRoute);
+  }
+
+  hookRoute() {
+    const { saved } = this.state;
```

Comment:
i don't know if you need to destructure this just for one variable. probably just reference `this.state.saved`